src/components/taskDetails/Details.tsx

```diff
@@ -1,20 +1,86 @@
-import React, { FC, useState } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import moment from 'moment';
 import { FaReceipt } from 'react-icons/fa6';
 import Link from 'next/link';
 import Tooltip from '@/components/common/Tooltip/Tooltip';
 import setColor from './taskPriorityColors';
 import extractRepoName from '@/utils/extractRepoName';
 import styles from './task-details.module.scss';
-import { TaskDetailsProps } from '@/interfaces/taskDetails.type';
+import {
+    DetailsContentProps,
+    TaskDetailsProps,
+} from '@/interfaces/taskDetails.type';
+import useUserData from '@/hooks/useUserData';
 import { STARTED_ON, ENDS_ON } from '@/constants/constants';
 
 type StringNumberOrUndefined = string | number | undefined;
 
-const Details: FC<TaskDetailsProps> = ({ detailType, value, url }) => {
+const DetailsContent: FC<DetailsContentProps> = ({
+    color,
+    isGitHubLink,
+    value,
+    gitHubIssueLink,
+    isTimeDetail,
+    formatDate,
+    tooltipActive,
+    renderedValue,
+    getRelativeTime,
+}) => {
+    return (
+        <span
+            className={styles.detailValue}
+            style={{ color: color ?? 'black' }}
+        >
+            {isGitHubLink && value ? (
+                <a
+                    className={styles.gitLink}
+                    href={gitHubIssueLink}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    aria-label="Open GitHub Issue"
+                    title={value}
+                >
+                    {isGitHubLink ? `${extractRepoName(value)}` : value}
+                </a>
+            ) : isTimeDetail ? (
+                <Tooltip
+                    content={formatDate(value)}
+                    tooltipPosition={{
+                        top: '-3.6rem',
+                        right: '-4.5rem',
+                    }}
+                >
+                    {tooltipActive ? formatDate(value) : getRelativeTime(value)}
+                </Tooltip>
+            ) : (
+                renderedValue
+            )}
+        </span>
+    );
+};
+
+const Details: FC<TaskDetailsProps> = (props) => {
+    const { detailType, value, url, isEditing, setEditedTaskDetails } = props;
     const color = value ? setColor?.[value] : undefined;
     const isGitHubLink = detailType === 'Link';
     const gitHubIssueLink = isGitHubLink ? value : undefined;
+    const [newEndOnDate, setNewEndOnDate] = useState('');
+    const { isUserAuthorized } = useUserData();
+
+    useEffect(() => {
+        if (!isEditing) setNewEndOnDate('');
+    }, [isEditing]);
+
+    const handleBlurOfEndsOn = () => {
+        const endsOn = new Date(`${newEndOnDate}`).getTime() / 1000;
+
+        if (endsOn > 0) {
+            setEditedTaskDetails?.((prev) => ({
+                ...prev,
+                endsOn,
+            }));
```

Comment:
You are using `useEffect` to clear the `newEndOnDate` when `isEditing` changes. This is fine but can lead to unnecessary re-renders. You can optimize the `useEffect` hook by ensuring that the value of `isEditing` is actually changing before resetting the state. Alternatively, you could consider a different way to reset the form state that doesn’t rely on this side effect.

Reply:
But I have already added a check if isEditiing is false only then, what I additionally can add is to check if newEndOnDate is not null.